src/app/reporting/Engine.cpp

```diff
@@ -178,7 +178,30 @@ DataModel::ActionReturnStatus RetrieveClusterData(DataModel::Provider * dataMode
     }
     else
     {
-        status = dataModel->ReadAttribute(readRequest, attributeValueEncoder);
+        ReadOnlyBufferBuilder<DataModel::AttributeEntry> attributes;
+        const ConcreteClusterPath clusterPath(readRequest.path.mEndpointId, readRequest.path.mClusterId);
+        status = dataModel->Attributes(clusterPath, attributes);
```

Comment:
Please talk to @Alami-Amine who is also working on this code.  We do need to do the existence checks here, but they need to come before the ACL check above, not at this point.